### PR TITLE
Jetpack: Add default spacing on form block variations

### PR DESCRIPTION
**Pull request by CGastrell**
This PR improves form (parent) block selection by adding a `style` section on all form variation blocks.

#### Changes proposed in this Pull Request:

* add a `style: { spacing: { padding: { ... } } }` as attribute on all form variation blocks

The change effectively adds a padding on the forms, making the form parent block more easily selectable. The padding is an option the user can later remove (hence, removing the ease of selection of the form), but it serves as a starting point while we figure out better ways to achieve the same result.

Alongside #26914 the form layout looks more consistent with the overall look n'feel of blocks in the editor.

Before:
image

After:
image

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
Insert a form on a post. See that the form block shows a Dimensions setting already set as padding: 16px:
image

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/contact-form/variations.js [hunk 1]
```diff
@@ -32,6 +32,18 @@ const variations = compact( [
 				},
 			],
 		],
+		attributes: {
```
Comment (jcheringer) [inline projects/plugins/jetpack/extensions/blocks/contact-form/variations.js:48]: Shouldn't we avoid this code duplication? 
A variable or the `map` function would do the job.

**Comment by digitalwaveride** [comment]
Nice Christian!

> making the form parent block more easily selectable.

As mentioned in our conversation yesterday, I generally really like any efforts to find ways for making it easier for users to discover and access a parent block.
Also, I like the general idea of this padding.

I have though concerns to create here a solution for a single block, which is essentially an issue that is much more general and impacts basically all parent/child block combinations.

The most general block that should give us likely direction for the best practice pattern behavior should be the group block.

I would recommend staying with the Group block behavior for now and asking for input from the Gutenberg channel, for any context for this interaction behavior.

**Review by jcheringer — APPROVED** [review]
Works for me!

**Comment by samiff** [comment]
> I would recommend staying with the Group block behavior for now and asking for input from the Gutenberg channel, for any context for this interaction behavior.

For clarity on this PR, do we want to hold off on merging for now?

**Comment by CGastrell** [comment]
> As mentioned in our conversation yesterday, I generally really like any efforts to find ways for making it easier for users to discover and access a parent block.
Also, I like the general idea of this padding.

> For clarity on this PR, do we want to hold off on merging for now?

@samiff As far as the first comment quote goes, I'd say that's a "merge it". The original purpose of this PR is to improve on the "looks" of the block while complying with the wysiwyg spirit of Gutenberg. The improvement on the block's parent selection is just a side effect, a "nice to have" that the user can actually ditch by removing the paddings from the sidebar.

I do stand by the fact that it looks better and I'd be willing to merge it (once the lock is lifted as there's a release in progress). The concerns about how the block's selection should work in general falls out of the scope of this PR and might be better addressed on a more _ad hoc_ discussion

**Review by samiff — APPROVED** [review]
Looks good to me 👍

**Pull request merged**